addons/frag/functions/fnc_dev_debugAmmo.sqf

```diff
@@ -15,67 +19,86 @@
  *
  * Public: No
  */
-
 params [
-    ["_debugMissing", true, [false]],
-    ["_debugForce", false, [false]],
-    ["_debugSkippedFragPower", 30, [0]]
+    ["_logAll", false, [false]],
+    ["_onlyShotAmmoTypes", false, [false]],
+    ["_csvFormat", false, [false]]
 ];
 
 diag_log text format ["~~~~~~~~~~~~~Start [%1]~~~~~~~~~~~~~", _this];
+if (_csvFormat) then {
+    diag_log text format ["ammo,gurney_c,gurney_m,gurney_k,gurney_gC,fragTypes,fragCount,Inheritance"];
+};
 
-private _allMagsConfigs = configProperties [configFile >> "CfgMagazines", "isClass _x", true];
-private _processedCfgAmmos = [];
-
-{
-    private _ammo = toLower getText (_x >> "ammo");
-    if (_ammo != "" && {!(_ammo in _processedCfgAmmos)}) then {
-        _processedCfgAmmos pushBack _ammo;
-
-        //Ignore mines/bombs
-        if (_ammo isKindOf "TimeBombCore") exitWith {};
+// Gather all configs, either those that could be created from firing or all classes
+private _allAmmoConfigs = [];
+if (_onlyShotAmmoTypes) then {
+    private _configSearchFunc = {
+        params [
+            ["_ammo", "", [""]]
+        ];
+        if (_ammo isEqualTo "" || {_ammo in _allAmmoConfigs}) exitWith {};
+        _allAmmoConfigs pushBack _ammo;
+        private _ammoConfig = configFile >> "cfgAmmo" >> _ammo;
+        private _subMunit = toLowerANSI getText (_ammoConfig >> "submunitionAmmo");
+        if (_subMunit isNotEqualTo "") then {
+            _subMunit = getArray (_ammoConfig >> "submunitionAmmo");
```

Comment:
`submunitionAmmo` can be either an array or a single classname, it was just a simplified and quick approach to grabbing the right type since the function `dev_debugAmmo` is meant to be a single shot developer funtction. I'll write it to be more efficient.

Reply:
Then my suggestion is the way to go imo.
First you see if it's an array. If yes, you get it as an array, if not, you `getText`. If `getText` returns `""`, skip/do nothing.

Reply:
Ah, I see, your suggestion before was vague and seemed like you just thought it was an array. As mentioned elsewhere, the function itself was a bit whack, fixed in next commit